src/main/java/com/conveyal/datatools/manager/gtfsplus/GtfsPlusValidation.java

```diff
@@ -122,6 +125,13 @@ private static void validateTable(
         GTFSFeed gtfsFeed
     ) throws IOException {
         String tableId = specTable.get("id").asText();
+        Boolean tableIsDirections = tableId.equals("directions");
+
+        Map<String, Route> gtfsRoutes = new HashMap<>();
+        if (tableIsDirections) {
+            // Copy the gtfs routes into a map we can "check them off" in (remove them)
+            gtfsRoutes.putAll(gtfsFeed.routes);
```

Comment:
Inline with the Set update, this should be `gtfsRoutes = gtfsFeed.routes.keySet();`

Reply:
Used `gtfsRoutes.addAll(gtfsFeed.routes.keySet())` to avoid reference issues with a copy

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/559/commits/cf83716376c894677cddf3752eb439962f1f4f62